### PR TITLE
fix: revert skipBuildPom default to always true for 'pom' packaging.

### LINKED ISSUE #1184: Unable to build multi-module project with fmp 3.5.34
### Description

When trying to build [this](https://github.com/snowdrop/spring-boot-configmap-booster/tree/sb-1.5.10) multi module project with fmp `3.5.34` I get the following error

```
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.34:build (fmp) on project spring-boot-configmap-parent: Failed to execute the build: Unable to create the build archive: Failed to create assembly for docker image  (with mode 'dir'): Error creating assembly archive docker: You must set at least one file.. If you include the build artifact please ensure that you have built the artifact before with 'mvn package' (should be available in the target/ dir). Please see the documentation (section "Assembly") for more information.
```

I should not that when building other non multi-module projects, everything worked just fine.

In order to bypass the problem, I downgraded to fmp `3.5.33`, where everything worked as expected

### Info

* f-m-p version : 3.5.34
* Maven version (`mvn -v`) : 3.5.2
```

```
* Kubernetes / OpenShift setup and version : Openshift 3.7
* If it's a bug, how to reproduce : Run mvn clean verify -Popenshift,openshift-it on the sb-1.5.10 brach

### Notes
You can easily switch versions of fmp in the project by setting the `fabric8-maven-plugin.version` Maven property

### PR DESCRIPTION
**Pull request by rhuss**
Fixes #1184 and reverts #1112

### CODE REVIEW

**Comment by rhuss** [comment]
@maxandersen this is the PR for reverting the behaviour. 

btw, I really wonder what the process now is for doing PR reviews and merges. 27 pending PRs dont look sane.

File: doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc [hunk 1]
```diff
@@ -177,7 +177,7 @@ a| The build mode which can be
 | `docker.skip.build`
 
 | *skipBuildPom*
-| If set the build step will be skipped for modules of type `pom`. If not set, then by default projects of type `pom` will be skipped if there are no image configurations contained.
+| If set to `false` the build step will not be skipped for modules of type `pom`. By default the plugin is not executed for modules with `pom` packaging.
 | `docker.skip.build.pom`
```
Comment (piyush1594) [inline doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc]: @rhuss Flag is fabric8.skip.build.pom if I am not wrong

**Comment by piyush1594** [comment]
@rhuss PR looks good to me. Let me test it locally against #1184

**Comment by piyush1594** [comment]
I just checked it locally and its working fine.

**Comment by maxandersen** [comment]
> btw, I really wonder what the process now is for doing PR reviews and merges. 27 pending PRs dont look sane.

what was the process before ? I think this is something to work out on https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!forum/fabric8-devclients with @piyush1594 and @pradeepto

File: doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc (hunk 1 above)
Comment (piyush1594) [reply doc/src/main/asciidoc/inc/goals/build/_fabric8-build.adoc]: Just this little change, else LGTM

**Comment by rhuss** [comment]
Updated, but actually, I don't like that all _skip_ parameters for builds are prefixed with `docker` (they are inherited from docker-maven-plugin) except this property which is prefixed with `fabric8`. Not very intuitive, but changing this of course is not backwards compatible as well.

**Comment by piyush1594** [comment]
@rhuss I checked 3.5.34 locally, it works fine for fabric8.skip.build.pom but it does not work for docker.skip.build.pom. I am not sure for previous versions.

**Comment by rhuss** [comment]
@piyush1594 @hrishin anything left to do for this PR to be approved ?

**Comment by piyush1594** [comment]
This looks good to me. Already approved.

**Pull request merged**